crates/sequencer/src/serde/contract.rs

```diff
@@ -0,0 +1,246 @@
+use starknet_api::deprecated_contract_class::{EntryPoint, EntryPointOffset, EntryPointType};
+
+use super::program::SerializableProgram;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum SerializableContractClass {
+    V0(SerializableContractClassV0),
+    V1(SerializableContractClassV1),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SerializableContractClassV0 {
+    pub program: SerializableProgram,
+    pub entry_points_by_type: HashMap<EntryPointType, Vec<SerializableEntryPoint>>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SerializableContractClassV1 {
+    pub program: SerializableProgram,
+    pub entry_points_by_type: HashMap<EntryPointType, Vec<SerializableEntryPointV1>>,
+    pub hints: HashMap<String, Vec<u8>>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SerializableEntryPoint {
+    pub selector: EntryPointSelector,
+    pub offset: SerializableEntryPointOffset,
+}
+
+impl From<EntryPoint> for SerializableEntryPoint {
+    fn from(value: EntryPoint) -> Self {
+        Self {
+            selector: value.selector,
+            offset: value.offset.into(),
+        }
+    }
+}
+
+impl From<SerializableEntryPoint> for EntryPoint {
+    fn from(value: SerializableEntryPoint) -> Self {
+        Self {
+            selector: value.selector,
+            offset: value.offset.into(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SerializableEntryPointOffset(pub usize);
+
+impl From<EntryPointOffset> for SerializableEntryPointOffset {
+    fn from(value: EntryPointOffset) -> Self {
+        Self(value.0)
+    }
+}
+
+impl From<SerializableEntryPointOffset> for EntryPointOffset {
+    fn from(value: SerializableEntryPointOffset) -> Self {
+        Self(value.0)
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SerializableEntryPointV1 {
+    pub selector: EntryPointSelector,
+    pub offset: SerializableEntryPointOffset,
+    pub builtins: Vec<String>,
+}
+
+impl From<SerializableEntryPointV1> for blockifier::execution::contract_class::EntryPointV1 {
+    fn from(value: SerializableEntryPointV1) -> Self {
+        blockifier::execution::contract_class::EntryPointV1 {
+            selector: value.selector,
+            offset: value.offset.into(),
+            builtins: value.builtins,
+        }
+    }
+}
+
+impl From<blockifier::execution::contract_class::EntryPointV1> for SerializableEntryPointV1 {
+    fn from(value: blockifier::execution::contract_class::EntryPointV1) -> Self {
+        SerializableEntryPointV1 {
+            selector: value.selector,
+            offset: value.offset.into(),
+            builtins: value.builtins,
+        }
+    }
+}
+
+impl TryFrom<SerializableContractClass> for blockifier::execution::contract_class::ContractClass {
+    type Error = anyhow::Error;
+
+    fn try_from(value: SerializableContractClass) -> Result<Self, Self::Error> {
```

Comment:
@greged93 this whole file is borrowed from Dojo, they said they are working on serialization for these types, if we still need this file in sometime, then might be we can refactor this otherwise we will probably not need these serde types in sometime!